src/components/Web3ConnectModal/Web3ConnectModal.tsx

```diff
@@ -25,14 +25,20 @@ export const Web3ConnectModal: React.FC<ModalProps> = ({
           <Web3ProviderButton
             image={MetamaskLogo}
             providerName="Metamask"
-            onClick={() => connect('injected')}
+            onClick={() => {
+              connect('injected');
+              localStorage.setItem('web3', 'injected');
```

Comment:
Never seen `localStorage` used, but looks promising. Interested to see how this behaves when switching wallets. Also there is `sessionStorage` which may be more appropriate, depending on behavior. Good PR. 

https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage